pkg/controller/instance/instance_controller.go

```diff
@@ -376,24 +414,46 @@ type ReconcileInstance struct {
 // +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=kudo.dev,resources=instances,verbs=get;list;watch;create;update;patch;delete
 func (r *ReconcileInstance) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	ctx := context.TODO()
 	// Fetch the Instance instance
-	instance := &kudov1alpha1.Instance{}
-	err := r.Get(context.TODO(), request.NamespacedName, instance)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			// Object not found, return.  Created objects are automatically garbage collected.
-			// For additional cleanup logic use finalizers.
-			return reconcile.Result{}, nil
-		}
-		// Error reading the object - requeue the request.
+	// Fetch the Instance instance
+	instance, err := getInstance(ctx, r, request)
```

Comment:
Similar to my comment above, can we break out the reconciler into a render method and an API interaction method?

Reply:
For example, this `Renconcile()` method could call `ReconcileInstance(instance *kudov1alpha.Instance) (*kudov1alpha.PlanExecution, error)`

Reply:
this is a significant improvement over what was there.   Moving the side effects out of predicates etc.   I expect there to be several more PRs to get this complete.   I like the direction of the comment / request but can we make this part of a future PR?    I feel this make good stride and is easily reviewable as a PR.

Reply:
Oh yes, of course! These are just suggestions while we're in here, I don't feel strongly about it.

Reply:
oh... I definitely want what you are asking for... it won't get lost.   I want to fix the instance controller to align with what a controller runtime controller is expect to be... then we need to refactor for encapsulation and testing!

Reply:
like that suggestion! That's what I did for instance to some extent and it improved the testability A LOT